alphagov/tsuru-ansible
# [#99121602] Configure elasticsearch

Story: [#99121602](https://www.pivotaltracker.com/n/projects/1275640/stories/99121602)

Configure elasticsearch server, deploy broker (runs on the same server) and register elasticsearch service in tsuru.

Depends on https://github.com/alphagov/tsuru-terraform/pull/86
### How to test
1. Run terraform from [this](https://github.com/alphagov/tsuru-terraform/pull/86) PR against your environment - this adds elasticsearch server
2. Run ansible from this PR - that configures the server, adds broker and registers service

To verify service was correctly added:
1. Create service in tsuru `tsuru service-add elasticsearch mysearch -t myteam`
2. Bind the service to your app `tsuru service-bind mysearch -a myapp`
3. Check environment variables of your app, there should be `ELASTICSEARCH_HOST` and `ELASTICSEARCH_PORT` private variables present.

To test the service itself, you can try deploying the [digital marketplace search api](https://github.com/RichardKnop/digitalmarketplace-search-api.git). After deploying the app, you can check https://<your_digitalmarketplace_search_api_app>/_status which should tell if all is fine and if any digitalmarketplace-search-api related data are present in elasticsearch.
### Who can test:

Anyone but @mtekel and @jimconner
